File: examples/university-demo/src/main/java/org/axonframework/examples/demo/university/faculty/events/StudentEnrolledInFaculty.java
package org.axonframework.examples.demo.university.faculty.events;

import org.axonframework.examples.demo.university.faculty.FacultyTags;
import org.axonframework.examples.demo.university.shared.ids.FacultyId;
import org.axonframework.examples.demo.university.shared.ids.StudentId;
import org.axonframework.eventsourcing.annotation.EventTag;

public record StudentEnrolledInFaculty(
        @EventTag(key = FacultyTags.FACULTY_ID)
        FacultyId facultyId,
        @EventTag(key = FacultyTags.STUDENT_ID)
        StudentId studentId,
        String firstName,
        String lastName
) {

}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/events/StudentEnrolledInFaculty.kt
package org.axonframework.examples.university.faculty.events

import org.axonframework.examples.university.faculty.FacultyTags
import org.axonframework.examples.university.shared.ids.StudentId
import org.axonframework.eventsourcing.annotation.EventTag

data class StudentEnrolledInFaculty(
    @EventTag(key = FacultyTags.STUDENT)
    val studentId: StudentId,
    val firstName: String,
    val lastName: String,
) : FacultyEvent
